Loop/Managers/StashManager/StashManager.swift

```diff
@@ -0,0 +1,504 @@
+
+        // Keep only one window as revealed
+        for revealedWindowId in store.revealed {
+            guard let revealedWindow = store.stashed[revealedWindowId] else { break }
+            hideWindow(revealedWindow, animate: animate)
+        }
+
+        let frame = window.computeRevealedFrame()
+
+        window.window.activate()
+        store.markWindowAsRevealed(window.id)
+        window.window.setFrame(frame, animate: animate)
+
+        print("StashManager: revealWindow \(window.window)")
+    }
+
+    /// Hides a stashed window by moving it to its stashed frame.
+    func hideWindow(_ window: StashedWindow, animate: Bool) {
+        guard !shouldThrottle(windowID: window.id) else { return }
+
+        let frame = window.computeStashedFrame(peekSize: stashedWindowVisiblePadding)
+
+        unfocus(window.id)
+        window.window.setFrame(frame, animate: animate)
+        store.markWindowAsHidden(window.id)
+
+        print("StashManager: hideWindow \(window.window)")
+    }
+
+    /// Checks if the window reveal / hide should be throttled based on the last reveal time.
+    func shouldThrottle(windowID: CGWindowID) -> Bool {
+        let now = Date.now
+        if let lastTime = lastRevealTime[windowID], now.timeIntervalSince(lastTime) < revealThrottleInterval {
+            return true
+        }
+        lastRevealTime[windowID] = now
+        return false
+    }
+
+    /// Attempts to unfocus (i.e., shift focus away from) a specified window.
+    ///
+    /// This method looks for the first (topmost) visible, non-minimized window on the same screen as the specified window,
+    /// and tries to activate it (i.e., bring it to the foreground).
+    func unfocus(_ windowID: CGWindowID) {
+        guard shiftFocusWhenStashed else { return }
+        guard let stashedWindow = store.stashed[windowID] else { return }
+        guard let screen = ScreenManager.screenContaining(stashedWindow.window) ?? NSScreen.main else { return }
+
+        let focusWindow = WindowEngine.windowList.first(where: { window in
+            guard let currentWindowScreen = ScreenManager.screenContaining(window) ?? NSScreen.main else { return false }
+            guard screen.isSameScreen(currentWindowScreen) else { return false }
+
+            return window.cgWindowID != windowID
+                && !window.isApplicationHidden
+                && !window.isWindowHidden
+                && !window.minimized
+        })
+
+        if let focusWindow {
+            print("StashManager: Focusing another window on the same screen: \(focusWindow).")
+            focusWindow.activate()
+        }
+    }
+}
+
+// MARK: - Mouse moved listener
+
+private extension StashManager {
+    func startListeningMouseMoved() {
+        guard mouseMonitor == nil else { return }
+
+        print("StashManager: Listening for mouse moved events…")
+
+        mouseMonitor = NSEventMonitor(scope: .global, eventMask: .mouseMoved) { [weak self] _ in
```

Comment:
```suggestion
        mouseMonitor = NSEventMonitor(scope: .all, eventMask: .mouseMoved) { [weak self] _ in
```

This will make stashing work even when Loop is the frontmost application.